Mathlib/Topology/Order/OrderClosedExtr.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2024 Bjørn Kjos-Hanssen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bjørn Kjos-Hanssen, Patrick Massot
+-/
+
+import Mathlib.Topology.Order.OrderClosed
+import Mathlib.Topology.Order.LocalExtr
+
+/-!
+# Local maxima from monotonicity and antitonicity
+
+In this file we prove a lemma that is useful for the First Derivative Test in calculus,
+and its dual.
+
+## Main statements
+
+* `isLocalMax_of_mono_anti` : if a function `f` is monotone to the left of `x`
+  and antitone to the right of `x` then `f` has a local maximum at `x`.
+
+* `isLocalMin_of_anti_mono` : the dual statement for minima.
+
+* `isLocalMax_of_mono_anti'` : a version of `isLocalMax_of_mono_anti` for filters.
+
+* `isLocalMin_of_anti_mono'` : a version of `isLocalMax_of_mono_anti'` for minima.
+
+-/
+
+open Set Topology Filter
+
+/-- If `f` is monotone on `(a,b]` and antitone on `[b,c)` then `f` has
+a local maximum at `b`. -/
+lemma isLocalMax_of_mono_anti.{u, v}
+    {α : Type u} [TopologicalSpace α] [LinearOrder α] [OrderClosedTopology α]
+    {β : Type v} [Preorder β]
+    {a b c : α} (g₀ : a < b) (g₁ : b < c) {f : α → β}
+    (h₀ : MonotoneOn f (Ioc a b))
+    (h₁ : AntitoneOn f (Ico b c)) : IsLocalMax f b :=
+  mem_of_superset (Ioo_mem_nhds g₀ g₁) (fun x _ => by rcases le_total x b <;> aesop)
+
+/-- If `f` is antitone on `(a,b]` and monotone on `[b,c)` then `f` has
+a local minimum at `b`. -/
+lemma isLocalMin_of_anti_mono.{u, v}
+    {α : Type u} [TopologicalSpace α] [LinearOrder α] [OrderClosedTopology α]
+    {β : Type v} [Preorder β] {a b c : α} (g₀ : a < b) (g₁ : b < c) {f : α → β}
+    (h₀ : AntitoneOn f (Ioc a b)) (h₁ : MonotoneOn f (Ico b c)) : IsLocalMin f b :=
+  mem_of_superset (Ioo_mem_nhds g₀ g₁) (fun x hx => by rcases le_total x b  <;> aesop)
+
+/-- If `L` and `R` are neighborhoods of `b` within sets whose union is `Set.univ`, then
+`L ∪ R` is a neighborhood of `b`. -/
+theorem union_mem_nhds_of_mem_nhdsWithin.{u} {α : Type u} [TopologicalSpace α] {b : α}
```

Comment:
This has nothing to do here, please move it next to ` mem_nhdsWithin_iff_exists_mem_nhds_inter`.

Reply:
OK, done.